core/bin/external_node/src/node_builder.rs

```diff
@@ -0,0 +1,511 @@
+        self = self
+            .add_l1_batch_commitment_mode_validation_layer()?
+            .add_validate_chain_ids_layer()?;
+
+        // Sort the components, so that the components they may depend on each other are added in the correct order.
+        components.sort_unstable_by_key(|component| match component {
+            // API consumes the resources provided by other layers (multiple ones), so it has to come the last.
+            Component::HttpApi | Component::WsApi => 1,
+            // Default priority.
+            _ => 0,
+        });
+
+        for component in &components {
+            match component {
+                Component::HttpApi => {
+                    self = self
+                        .add_sync_state_updater_layer()?
+                        .add_api_caches_layer()?
+                        .add_tree_api_client_layer()?
+                        .add_main_node_fee_params_fetcher_layer()?
+                        .add_tx_sender_layer()?
+                        .add_http_web3_api_layer()?;
+                }
+                Component::WsApi => {
+                    self = self
+                        .add_sync_state_updater_layer()?
+                        .add_api_caches_layer()?
+                        .add_tree_api_client_layer()?
+                        .add_main_node_fee_params_fetcher_layer()?
+                        .add_tx_sender_layer()?
+                        .add_ws_web3_api_layer()?;
+                }
+                Component::Tree => {
+                    // Right now, distributed mode for EN is not fully supported, e.g. there are some
+                    // issues with reorg detection and snapshot recovery.
+                    // So we require the core component to be present, e.g. forcing the EN to run in a monolithic mode.
+                    anyhow::ensure!(
+                        components.contains(&Component::Core),
+                        "Tree must run on the same machine as Core"
+                    );
+                    let with_tree_api = components.contains(&Component::TreeApi);
+                    self = self.add_metadata_calculator_layer(with_tree_api)?;
+                }
+                Component::TreeApi => {
+                    anyhow::ensure!(
+                        components.contains(&Component::Tree),
+                        "Merkle tree API cannot be started without a tree component"
+                    );
+                    // Do nothing, will be handled by the `Tree` component.
+                }
+                Component::TreeFetcher => {
+                    self = self.add_tree_data_fetcher_layer()?;
+                }
+                Component::Core => {
+                    // Core is a singleton & mandatory component,
+                    // so until we have a dedicated component for "auxiliary" tasks,
+                    // it's responsible for things like metrics.
+                    self = self.add_postgres_metrics_layer()?;
```

Comment:
Not fully understand why we decided to use `add_postgres_metrics_layer` only if `Component::Core` is enabled?
Orienting on the current initialization(`core/bin/external_node/src/main.rs  line:913`), we add `metrics_task` before `init_tasks`. Also, I see that the pool is used not only by the `Core` component.

Reply:
Old EN codebase is not fully adapted for working in a distributed way. Postgres metrics task scrapes the metrics from postgres -- they will be the same for any component that is connected to postgres -- and exports them to prometheus. We only need a single component exporting these metrics, because otherwise the exported metrics will be duplicated.

So 2 parts of the answer for your question:
1. We need to run it with core, because we need it to run only with a single component, and `Core` is a mandatory singleton componenet of EN. On main node, we have house keeper for such stuff.
2. The old code was written before EN could run in a distributed mode, and it is not suitable for distributed mode.

Reply:
that answers my question. thanks!